app/factory/Layer.js

```diff
@@ -78,6 +78,13 @@ Ext.define('CpsiMapview.factory.Layer', {
             //do nothing, and return empty layer
         }
 
+        // TODO: make sure this listener is called only once
+
+        // listener that checks the resolution and
+        // changes the switch layer if required
+        var olMap = BasiGX.util.Map.getMapComponent().getMap();
+        olMap.getView().on('change:resolution', LayerFactory.handleSwitchLayerOnResolutionChange);
```

Comment:
I am not sure if this listener gets created once or multiple times. The performance seems to fine. But it would be cleaner to make sure that listener is created only once.

Reply:
Yes it is bound multiple times, I'll come up with a change for that.

Reply:
See cf91b4c